packages/mux-player/src/index.ts

```diff
@@ -141,6 +141,98 @@ class MuxPlayerElement extends VideoApiElement {
+    this.#setUpCaptionsMovement();
+  }
+
+  #setUpCaptionsMovement() {
+    const mc = this.shadowRoot.querySelector('media-controller');
+    if (!mc) return;
+
+    let selectedTrack;
+    let cuesmap = new Map();
+
+    // toggles activeCues for a particular track depending on whether the user is active or not
+    const toggleLines = (track, userInactive) => {
+      const cues = (track && track.activeCues) || [];
+
+      Array.from(cues).forEach((cue) => {
+        // ignore cues that are
+        // - positioned vertically via percentage.
+        // - cues that are not at the bottom
+        //   - line is less than -5
+        //   - line is between 0 and 10
+        if (!cue.snapToLines || cue.line < -5 || (cue.line > 0 && cue.line < 10)) {
+          return;
+        }
+
+        // if the user is active or if the player is paused, the captions should be moved up
+        if (!userInactive || this.paused) {
+          // for cues that have more than one line, we want to push the cue further up
+          const lines = cue.text.split('\n').length;
+          // start at -3 to account for thumbnails as well.
+          const setTo = -3 - lines;
+
+          // if the line is already set to -4, we don't want to update it again
+          // this can happen in the same tick on chrome and safari which fire a cuechange
+          // event when the line property is changed to a different value.
+          if (cue.line === setTo) {
+            return;
+          }
+
+          if (!cuesmap.has(cue)) {
+            cuesmap.set(cue, cue.line);
+          }
+
+          // we have to set line to 0 first due to a chrome bug https://crbug.com/1308892
+          cue.line = 0;
+          cue.line = setTo;
+        } else {
+          cue.line = cuesmap.get(cue) || 'auto';
+        }
+      });
+    };
+
+    // this is necessary so that if a cue becomes active while the user is active, we still position it above the control bar
+    const cuechangeHandler = () => {
+      if (mc.hasAttribute('user-inactive')) {
+        return;
+      }
+
+      setTimeout(() => toggleLines(selectedTrack, false), 100);
+    };
+
+    const selectTrack = () => {
+      const newSelectedTrack = Array.from(mc?.media?.textTracks || []).filter(
+        (t) => ['subtitles', 'captions'].includes(t.kind) && t.mode === 'showing'
+      )[0];
+
+      if (newSelectedTrack !== selectedTrack) {
+        selectedTrack?.removeEventListener('cuechange', cuechangeHandler);
+      }
+
+      selectedTrack = newSelectedTrack;
+      selectedTrack?.addEventListener('cuechange', cuechangeHandler);
+      // selectedTrack?.addEventListener('cuechange', () => {mc.pause()}, {once: true});
+    };
+
+    selectTrack();
+    // update the selected track as necessary
+    mc?.media?.textTracks.addEventListener('change', selectTrack);
+    mc?.media?.textTracks.addEventListener('addtrack', selectTrack);
+
+    mc?.addEventListener('userinactivechange', () => {
```

Comment:
this depends on a media-chrome release with https://github.com/muxinc/media-chrome/pull/182 and https://github.com/muxinc/media-chrome/pull/190